erpnext/maintenance/doctype/maintenance_schedule/maintenance_schedule.js

```diff
@@ -54,6 +54,23 @@ erpnext.maintenance.MaintenanceSchedule = frappe.ui.form.Controller.extend({
 	contact_person: function() {
 		return erpnext.utils.get_contact_details(this.frm);
 	},
+
+	create_opportunity: function(doc, cdt, cdn) {
+		var me = this;
+		frappe.call({
+			method: "erpnext.maintenance.doctype.maintenance_schedule.maintenance_schedule.create_maintenance_opportunity",
+			args: {
+				maintenance_schedule: doc.name,
+				row: cdn
+			},
+			callback: function(r) {
+				if (!r.exc) {
+					frappe.msgprint(__("Opportunity created successfully."));
```

Comment:
Do not create and save maintenance Opportunity .. just make document object, return to API caller and sync and navigate to local form. See `claim_items_to_be_billed.js` line 240 for reference